nixos/modules/services/backup/rsnapshot.nix

```diff
@@ -60,13 +62,28 @@ in
         cmd_ssh	${openssh}/bin/ssh
         cmd_logger	${inetutils}/bin/logger
         cmd_du	${coreutils}/bin/du
-        lockfile	/run/rsnapshot.pid
+        lockfile	${rundir}/rsnapshot.pid
 
         ${cfg.extraConfig}
       '');
     in {
       environment.systemPackages = [ myRsnapshot ];
 
+      systemd.services.rsnapshot.preStart = ''
```

Comment:
A little simpler :D Drop that function, always do mkdir -m -p and chmod and you're done without all those tests.